CRM/Mailing/Event/BAO/Reply.php

```diff
@@ -156,9 +161,12 @@ public static function send($queue_id, &$mailing, &$bodyTxt, $replyto, &$bodyHTM
       $h = $parsed->headers->getCaseSensitiveArray();
       $b = $parsed->generateBody();
 
-      // strip Return-Path of possible bounding brackets, CRM-4502
+      // Rewrite any invalid Return-Path headers.
       if (!empty($h['Return-Path'])) {
-        $h['Return-Path'] = trim($h['Return-Path'], '<>');
+        $h['Return-Path'] = $fromEmail;
+      }
+      if (!empty($h['Return-path'])) {
```

Comment:
@mfb any reason why we have the two versions of return path?

Reply:
This functionality is forwarding arbitrary emails received from the outside world, including headers.  And the email header storage is case-sensitive.  These emails could contain arbitrary headers such as `Return-Path` or `Return-path` - hopefully no other case combinations are ever seen, because they would not be supported by this code.

Reply:
@mfb can you use ```else``` instead of ```if``` in here?

Reply:
on PHP >= 5.6  we could array_filter by key but I think Civi still supports PHP 5.5 at the moment.

Reply:
Hmm actually I think I should rewrite this, because it looks like up above there are case-insensitive headers available!

Reply:
Ok I think this should be resolved. We can simply do `$parsed->setHeader('Return-Path', $fromEmail);` and rely on the library to merge in the header value we want regardless of case.